neo/Consensus/ConsensusService.cs

```diff
@@ -85,10 +85,17 @@ private bool CheckPolicy(Transaction tx)
             return true;
         }
 
-        private void CheckSignatures()
+        private void OnCommitAgreement(ConsensusPayload payload, CommitAgreement message)
         {
+            Log($"{nameof(OnCommitAgreement)}: height={payload.BlockIndex} hash={message.BlockHash.ToString()} view={message.ViewNumber} index={payload.ValidatorIndex}");
+
+            if (context.State.HasFlag(ConsensusState.BlockSent) ||
+                !context.TryToCommit(payload, message)) return;
+
             if (context.Signatures.Count(p => p != null) >= context.M && context.TransactionHashes.All(p => context.Transactions.ContainsKey(p)))
```

Comment:
Maybe add `context.State.HasFlag(ConsensusState.CommitSent)` here in this last verification before relay.

Reply:
I mean line 95.

Reply:
But this method is called when receive a commit, no when you sent your commit, so if you receive 6 commits, and you don't send your commit, you never relay the block.